### PR TITLE
Improve license script

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: tools/licenses/lib/main.dart [hunk 1]
```diff
@@ -357,37 +357,68 @@ class RepositoryBlankLicenseFile extends RepositorySingleLicenseFile {
...
+    r' *You may obtain a copy of the License at\n'
+    r' *\n'
+    r' *(http://www\.apache\.org/licenses/LICENSE-2\.0)\n'
+    r' *\n'
+    r' *Unless required by applicable law or agreed to in writing, software\n'
+    r' *distributed under the License is distributed on an "AS IS" BASIS,\n'
+    r' *WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied\.\n'
+    r' *See the License for the specific language governing permissions and\n'
+    r' *limitations under the License\.\n',
+    multiLine: true,
```
Comment (Hixie) [inline tools/licenses/lib/main.dart]: why doesn't the usual automatic apache license detection work here?

File: tools/licenses/lib/main.dart [hunk 2]
```diff
@@ -357,37 +357,68 @@ class RepositoryBlankLicenseFile extends RepositorySingleLicenseFile {
...
-      throw 'unexpected okhttp license file contents';
-    return new License.fromUrl(match.group(2), origin: io.fullName);
+    if (match == null || match.groupCount != 1)
+      throw 'unexpected apiclient license file contents';
+    return new License.fromUrl(match.group(1), origin: io.fullName);
+  }
+
+  @override
+  License licenseOfType(LicenseType type) {
+    return null;
```
Comment (Hixie) [inline tools/licenses/lib/main.dart]: should this return `this` for apache?

File: tools/licenses/lib/main.dart [hunk 3]
```diff
@@ -357,37 +357,68 @@ class RepositoryBlankLicenseFile extends RepositorySingleLicenseFile {
...
+  }
+}
+
+class RepositoryCatapultCoverageLicenseFile extends RepositorySingleLicenseFile {
```
Comment (Hixie) [inline tools/licenses/lib/main.dart]: similar questions on this class

**Comment by Hixie** [comment]
LGTM if there's a good reason this directory needs special-casing.

File: tools/licenses/lib/main.dart (hunk 1 above)
Comment (abarth) [reply tools/licenses/lib/main.dart]: It's the same situation as okhttp used to have.  This file doesn't contain the full apache license.  Just this shorter text and a link to the full license.

**Comment by chinmaygarde** [comment]
Can this be committed?

**Comment by abarth** [comment]
Done.

**Pull request merged**